cmake/Dependencies.cmake

```diff
@@ -41,45 +41,91 @@ if(BUILD_CAFFE2)
 endif()
 
 # ---[ BLAS
-if(BUILD_ATEN)
-  set(BLAS "MKL" CACHE STRING "Selected BLAS library")
-else()
-  set(BLAS "Eigen" CACHE STRING "Selected BLAS library")
+if(NOT BLAS_SET_BY_USER)
+  if(BUILD_ATEN)
+    set(BLAS "MKL" CACHE STRING "Selected BLAS library")
+  else()
+    set(BLAS "Eigen" CACHE STRING "Selected BLAS library")
+  endif()
 endif()
 set_property(CACHE BLAS PROPERTY STRINGS "Eigen;ATLAS;OpenBLAS;MKL;vecLib")
 message(STATUS "The BLAS backend of choice:" ${BLAS})
 
-if(BLAS STREQUAL "Eigen")
-  # Eigen is header-only and we do not have any dependent libraries
-  set(CAFFE2_USE_EIGEN_FOR_BLAS ON)
-elseif(BLAS STREQUAL "ATLAS")
-  find_package(Atlas REQUIRED)
-  include_directories(${ATLAS_INCLUDE_DIRS})
-  list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS ${ATLAS_LIBRARIES})
-  list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS cblas)
+# Atlas
+if(BLAS STREQUAL "ATLAS")
+  if(BLAS_SET_BY_USER)
+    find_package(Atlas REQUIRED)
+  else()
+    find_package(Atlas)
+  endif()
+  if(Atlas_FOUND)
```

Comment:
This can be put into the `else()` branch.

Reply:
The include_directories list-appends have to go after both find_package calls though. I could move if(NOT Atlas_FOUND) with a 'break' into the else(), but that doesn't seem worth it to me